src/types/num.rs

```diff
@@ -10,3 +13,299 @@ use crate::{ffi, PyAny};
 pub struct PyLong(PyAny);
 
 pyobject_native_type_core!(PyLong, pyobject_native_static_type_object!(ffi::PyLong_Type), #checkfunction=ffi::PyLong_Check);
+
+impl PyLong {
+    /// Creates a new `PyLong` from a C `long` integer.
+    /// # Returns
+    /// A [`Bound`] reference to a `PyLong` object representing the given `val`.
+    #[inline]
+    pub fn new_bound_from_c_long(py: Python<'_>, val: c_long) -> Bound<'_, Self> {
+        unsafe {
+            ffi::PyLong_FromLong(val)
+                .assume_owned(py)
+                .downcast_into_unchecked()
+        }
+    }
+}
+
+/// Implementation of functionality for [`PyLong`].
+///
+/// These methods are defined for the `Bound<'py, PyLong>` smart pointer, so to use method call
+/// syntax these methods are separated into a trait, because stable Rust does not yet support
+/// `arbitrary_self_types`.
+#[doc(alias = "PyLong")]
+pub trait PyLongMethods<'py>: crate::sealed::Sealed {
+    /// Gets the value of this int.
+    fn value(&self) -> c_long;
+}
+
+impl<'py> PyLongMethods<'py> for Bound<'py, PyLong> {
+    fn value(&self) -> c_long {
+        unsafe {
+            // Safety: self is PyLong object
+            ffi::PyLong_AsLong(self.as_ptr())
+        }
+    }
+}
```

Comment:
This implementation is incorrect if its value exceeds what `c_long` can hold (which is only an i32 on Windows). Also, I don't think having this implementation is a good idea.

Reply:
When you say its not a good idea to have this implementation, you mean having the `.value()` method?

I based my implementation heavily on `PyFloat`, which is why I wanted to clarify.

Reply:
It makes sense for PyFloat to have this method, because all possible values of a PyFloat can be expressed as a floating point number. But Python's integer has unlimited precision/size, so not all Python integers can be expressed with `c_long` (nor any other rust integer type like u64 or i128 for example).

Reply:
Got it! I agree! Removed the value method!